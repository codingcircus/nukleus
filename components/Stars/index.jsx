import React, {Component, PropTypes} from 'react';

function roundToHalves (number) {
  return Math.round(number * 2) / 2;
}

function getClassFrom (value) {
  return `rating-${roundToHalves(value).toString().replace('.', '-')}`;
}

export default class Stars extends Component {
  static propTypes = {
    value: PropTypes.number.isRequired
  };

  constructor ({value}) {
    super();
    this.starValue = getClassFrom(value);
    this.value = Math.round(value * 100) / 100;
  }

  render () {
    return (
      <rating className="rating rate-inline">
        <div className={this.starValue}>
          <span className="sr-only">Rating {this.value}</span>
          {[...Array(5)].map((item, key) =>
            <i
              key={key}
              className={`rating-star rating-star-${key + 1} fa`} />
          )}
        </div>
      </rating>
    );
  }
}
