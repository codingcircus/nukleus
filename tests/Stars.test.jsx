import React from 'react';
import renderer from 'react-test-renderer';
import Stars from 'Stars'; // eslint-disable-line import/no-unresolved, import/extensions, import/no-extraneous-dependencies

test('Renders Stars without crashing', () => {
  const component = renderer.create(
    <Stars value={2.5} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
