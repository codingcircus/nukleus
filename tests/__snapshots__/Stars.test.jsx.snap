exports[`test Renders Stars without crashing 1`] = `
<rating
  className="rating rate-inline">
  <div
    className="rating-2-5">
    <span
      className="sr-only">
      Rating 
      2.5
    </span>
    <i
      className="rating-star rating-star-1 fa" />
    <i
      className="rating-star rating-star-2 fa" />
    <i
      className="rating-star rating-star-3 fa" />
    <i
      className="rating-star rating-star-4 fa" />
    <i
      className="rating-star rating-star-5 fa" />
  </div>
</rating>
`;
